File: tracking-program/8087/api/models/MeterDataAggregate.js
/**
 * MeterDataAggregate.js
 *
 * @description :: A set of data points representing a single day's worth of meter readings for a project.
 */

module.exports = {

  attributes: {

    // The project that this data is for.
    project: {
      model: 'Project',
      required: true
    },

    // The day that the data is for, e.g. '2017-05-01'.
    day: {
      type: 'string',
      required: true
    },

    // The 15 minute interval during which this aggregate is for, including time zone offset.
    // Example: '0/-6:00', '25/+2:00', '95/-5:00'.
    // If blank, the aggregate is for the entire day.
    intervalId: {
      type: 'string',
    },

    // The number of samples included in this aggregate.
    numSamples: {
      type: 'number',
      required: true
    },

    // The timestamp marking the start of this interval period.
    intervalStartTime: {
      type: 'number'
    },

    // The timestamp marking the start of this interval period.
    intervalEndTime: {
      type: 'number'
    },

    //  ┌─┐┌─┐┬ ┬┌─┐┬─┐  ┬─┐┌─┐┌─┐┌┬┐┬┌┐┌┌─┐┌─┐
    //  ├─┘│ ││││├┤ ├┬┘  ├┬┘├┤ ├─┤ │││││││ ┬└─┐
    //  ┴  └─┘└┴┘└─┘┴└─  ┴└─└─┘┴ ┴─┴┘┴┘└┘└─┘└─┘

    // The AVERAGE of all voltage readings for the interval/day.
    avgVolt: {
      type: 'number'
    },

    // The AVERAGE of all amperage readings for the interval/day.
    avgAmp: {
      type: 'number'
    },

    // The AVG of all KW readings for the interval/day.
    avgKw: {
      type: 'number'
    },

    // The SUM of all Kva readings for the interval/day.
    avgKva: {
      type: 'number'
    },

    // The AVERAGE of all power factor readings for the interval/day.
    avgPf: {
      type: 'number'
    },

    // The AVERAGE of all Kvar readings for the interval/day.
    avgKvar: {
      type: 'number'
    },

    // The PEAK KVA reading across the entire day (only valid for daily rollup).
    peakKva: {
      type: 'number',
      allowNull: true
    },

    // The PEAK KW reading across the entire day (only valid for daily rollup).
    peakKw: {
      type: 'number',
      allowNull: true
    },

    multiplier: {
      type: 'number',
      allowNull: true
    }

  },

};

File: tracking-program/8087/flask_app/app/models/meter_data_aggregate.py
"""MeterDataAggregate model - api/models/MeterDataAggregate.js"""
from app.extensions import db
from app.models.base import BaseModel


class MeterDataAggregate(BaseModel):
    __tablename__ = "meterdataaggregate"

    project = db.Column(db.Integer, db.ForeignKey("project.id"), nullable=False)
    day = db.Column(db.String(255), nullable=False)
    intervalId = db.Column(db.String(255))
    numSamples = db.Column(db.Integer, nullable=False)
    intervalStartTime = db.Column(db.BigInteger)
    intervalEndTime = db.Column(db.BigInteger)
    avgVolt = db.Column(db.Float)
    avgAmp = db.Column(db.Float)
    avgKw = db.Column(db.Float)
    avgKva = db.Column(db.Float)
    avgPf = db.Column(db.Float)
    avgKvar = db.Column(db.Float)
    peakKva = db.Column(db.Float, nullable=True)
    peakKw = db.Column(db.Float, nullable=True)
    multiplier = db.Column(db.Float, nullable=True)
